openedx/features/course_experience/tests/test_url_helpers.py

```diff
@@ -0,0 +1,227 @@
+"""
+Test some of the functions in url_helpers
+"""
+from unittest import mock
+
+import ddt
+
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+
+from .. import url_helpers
+
+
+def _patch_courseware_mfe_is_active(ret_val):
+    return mock.patch.object(
+        url_helpers,
+        'courseware_mfe_is_active',
+        return_value=ret_val,
+    )
+
+
+@ddt.ddt
+class GetCoursewareUrlTests(SharedModuleStoreTestCase):
+    """
+    Test get_courseware_url.
+
+    Mock out `courseware_mfe_is_active`; that is tested elseware.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Set up data used across test functions.
+
+        We build two simple course structures (one using Split, the other Old Mongo).
+        Each course structure is a non-branching tree from the root Course block down
+        to the Component-level problem block; that is, we make one item for each course
+        hierarchy level.
+
+        For easy access in the test functions, we expose them in `self.items`
+        in a dict like this:
+        {
+            "split": {
+                "course_run": <course block for Split Mongo course>,
+                "section": <chapter block in course run>
+                "subsection": <sequence block in section>
+                "unit": <vertical block in subsection>
+                "component": <problem block in unit>
+            },
+            "mongo": {
+                "course_run": <course block for (deprecated) Old Mongo course>,
+                ... etc ...
+            }
+        }
+        """
+        super().setUpClass()
```

Comment:
You shouldn't need to run `setUpClass()` if you're running `setUpClassAndTestData()`

Reply:
(This weirdness was done because of ordering issues–to make the class play well with Django's `setUpTestData`, which wasn't a thing when `SharedModuleStoreTestCase` was first created.)

Reply:
Good to know :+1: